provisioning/provisioning-device-client/src/main/java/com/microsoft/azure/sdk/iot/provisioning/device/internal/contract/amqp/AmqpsProvisioningSaslHandler.java

```diff
@@ -237,12 +237,24 @@ public void handleOutcome(SaslOutcome outcome) throws ProvisioningDeviceSecurity
         }
     }
 
+    @Override
+    public String plainUsername()
+    {
+        throw new UnsupportedOperationException("TPM sasl does not use plain mechanism for authentication");
+    }
+
+    @Override
+    public String plainPassword()
+    {
+        throw new UnsupportedOperationException("TPM sasl does not use plain mechanism for authentication");
+    }
+
     /**
      * Sets the value of this object's saved sas token. Should only be called when that sas token was generated using the
      * nonce retrieved from the Device Provisioning service.
      * @param sasToken The SAS token to be used when finishing Sasl negotiation.
      */
-    void setSasToken(String sasToken)
+    public void setSasToken(String sasToken)
     {
```

Comment:
You know the question here. Why public ? :)

Reply:
The setSasToken function became part of the SaslHandler interface, so it had to become public. Interfaces don't allow for protected/private/etc.